### PR TITLE
[ci_yaml] Add autoroller builder

### PR DESCRIPTION
**Pull request by CaseyHillers**
https://github.com/flutter/flutter/issues/84998

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

### CODE REVIEW

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by CaseyHillers** [comment]
Manual roll CL at https://flutter-review.googlesource.com/c/infra/+/16280

File: .ci.yaml [hunk 1]
```diff
@@ -319,3 +319,10 @@ targets:
       jazzy_version: 0.9.5
     timeout: 90
     scheduler: luci
+
+  - name: Linux ci_yaml engine roller
```
Comment (keyonghan) [inline .ci.yaml:323]: Shall we add a `run_if`? Run this builder only when `.ci.yaml` changes. This way no bot resource is needed for most PRs.

Comment (CaseyHillers) [reply .ci.yaml:323]: This is postsubmit only, and we currently do not support run_if on prod tasks. This is using linux vms so it shouldn't be expensive.

**Pull request merged**